workflows/go/sdk/order-processor/workflow.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/dapr/go-sdk/client"
+	"github.com/dapr/go-sdk/workflow"
+)
+
+// OrderProcessingWorkflow is the main workflow for orchestrating activities in the order process.
+func OrderProcessingWorkflow(ctx *workflow.WorkflowContext) (any, error) {
+	orderID := ctx.InstanceID()
+	var orderPayload OrderPayload
+	if err := ctx.GetInput(&orderPayload); err != nil {
+		return nil, err
+	}
+	if err := ctx.CallActivity(NotifyActivity, workflow.ActivityInput(Notification{Message: fmt.Sprintf("Received order %s for %d %s - $%d", orderID, orderPayload.Quantity, orderPayload.ItemName, orderPayload.TotalCost)})).Await(nil); err != nil {
+		return OrderResult{Processed: false}, err
+	}
+
+	var verifyInventoryResult InventoryResult
+	if err := ctx.CallActivity(VerifyInventoryActivity, workflow.ActivityInput(InventoryRequest{
+		RequestID: orderID,
+		ItemName:  orderPayload.ItemName,
+		Quantity:  orderPayload.Quantity,
+	})).Await(&verifyInventoryResult); err != nil {
+		return OrderResult{Processed: false}, err
+	}
+
+	if !verifyInventoryResult.Success {
+		if err := ctx.CallActivity(NotifyActivity, workflow.ActivityInput(Notification{Message: fmt.Sprintf("Insufficient inventory for %s", orderPayload.ItemName)})).Await(nil); err != nil {
+			return OrderResult{Processed: false}, err
+		}
+	}
+
+	if orderPayload.TotalCost > 50000 {
+		var approvalRequired ApprovalRequired
+		if err := ctx.CallActivity(RequestApprovalActivity, workflow.ActivityInput(orderPayload)).Await(&approvalRequired); err != nil {
+			return OrderResult{Processed: false}, err
+		}
+		if err := ctx.WaitForExternalEvent("manager_approval", time.Second*200).Await(nil); err != nil {
+			return OrderResult{Processed: false}, err
+		}
+		// TODO: Confirm timeout flow - this will be in the form of an error.
+		if approvalRequired.Approval {
+			if err := ctx.CallActivity(NotifyActivity, workflow.ActivityInput(Notification{Message: fmt.Sprintf("Payment for order %s has been approved!", orderID)})).Await(nil); err != nil {
+				log.Printf("failed to notify of a successful order: %v\n", err)
+			}
+		} else {
+			if err := ctx.CallActivity(NotifyActivity, workflow.ActivityInput(Notification{Message: fmt.Sprintf("Payment for order %s has been rejected!", orderID)})).Await(nil); err != nil {
+				log.Printf("failed to notify of an unsuccessful order :%v\n", err)
+			}
+			return OrderResult{Processed: false}, nil
+		}
+	}
+	if err := ctx.CallActivity(ProcessPaymentActivity, workflow.ActivityInput(PaymentRequest{
+		RequestID:          orderID,
+		ItemBeingPurchased: orderPayload.ItemName,
+		Amount:             orderPayload.TotalCost,
+		Quantity:           orderPayload.Quantity,
+	})).Await(nil); err != nil {
```

Comment:
This is a bit of a personal preference (and maybe a nit), but i find multi-line code within an `if` statement to be very hard to read


```suggestion
    err := ctx.
        CallActivity(ProcessPaymentActivity, workflow.ActivityInput(PaymentRequest{
            RequestID:          orderID,
            ItemBeingPurchased: orderPayload.ItemName,
            Amount:             orderPayload.TotalCost,
            Quantity:           orderPayload.Quantity,
        })).
        Await(nil)
	if err != nil {
```